### ISSUE TITLE
Using the merge in the the Keras functional API

### ISSUE THREAD

**Issue by alyato** [fchollet/keras#4280]
I want to use the  Keras functional API to train my model,but it show me an error.
![cc](https://cloud.githubusercontent.com/assets/6815071/19968078/f050c3e0-a20d-11e6-9794-af30580ecef7.png)

And the error is 
![ff](https://cloud.githubusercontent.com/assets/6815071/19968105/161ac332-a20e-11e6-8534-71919a482e64.png)

I don't know how to solve it ? Thanks.

**Comment by CCXD** [comment]
> Users may pass data as a list of arrays, dictionary of arrays,
>     or as a single array. We normalize this to an ordered list of
>     arrays (same order as `names`), while checking that the provided
>     arrays have shapes that match the network's expectations.

What's the type of X_train and Y_train?

**Comment by alyato** [comment]
@CCXD,Thanks. i show post the load_data:
![ww](https://cloud.githubusercontent.com/assets/6815071/19991474/977e7524-a270-11e6-8c53-f832393b72b2.png)

I test it,using the `validation_split`
`model.fit(X_train, Y_train, batch_size=batch_size, nb_epoch=nb_epoch,verbose=1, validation_split=0.8)`
it works.
when i set `validation_data=(X_train,Y_train)`,it also works.
`model.fit(X_train, Y_train, batch_size=batch_size, nb_epoch=nb_epoch,verbose=1, validation_data=(X_train,Y_train))`

**Comment by alyato** [comment]
@CCXD 
the shape of `X_train` is `(4000, 3, 48, 48)`
the shape of  `Y_train` is `(4000, 8)`
and
the shape of `m` is `(470, 3, 48, 48)`
the shape of  `n` is  `(470, 8)`

**Comment by alyato** [comment]
i solve it by following the standard name,not using the random naming such as m,or n
